aiohttp/web_urldispatcher.py

```diff
@@ -174,15 +175,15 @@ def __init__(
         if expect_handler is None:
             expect_handler = _default_expect_handler
 
-        assert asyncio.iscoroutinefunction(
+        assert inspect.iscoroutinefunction(
             expect_handler
         ), f"Coroutine is expected, got {expect_handler!r}"
```

Comment:
As per the discussion in that thread, these don't operate exactly the same, so we should probably retain backwards compatibility with undeprecated releases (especially as <3.12 can't even use the new method of marking sync functions).

```suggestion
        assert inspect.iscoroutinefunction(expect_handler) or (sys.version_info < (3, 14) and asyncio.iscoroutinefunction(expect_handler)), f"Coroutine is expected, got {expect_handler!r}"
```

Reply:
We can probably leave the pytest_plugin though, as it's test code and highly unlikely someone would use a wrapped sync function as a test function.